### PR TITLE
Enable every Aspire AppHost language

### PR DESCRIPTION
**Pull request by IEvangelist**
Stack 5/5. Depends on #1637.

This final activation change flips the four shared registry bits that publish Python, Go, Java, and Rust everywhere automatically. All four languages remain marked **Experimental**, and the single-registry-bit contract provides a one-bit rollback or toggle.

The activated registry now drives selectors, tabs, pivots, sidebar navigation, search, AppHost API routes and Markdown, integrations gallery links, project and sample pages, media, builder surfaces, homepage content, sample metadata, and container labels. `SimpleAppHostCode` continues to render explicit limitations for unauthored language variants instead of fallback code.

Validation:

- 548 unit tests passed in the validated final state; the requested focused activation run passed 252 tests across apphost-languages, apphost-language-coverage, apphost-language-docs-loader, integration parity, diagnostics-language, api-markdown, update-samples, and custom-components.
- Full twoslash validation passed.
- Lint and Astro sync passed.
- AppHost API generator validation passed 36/36.
- 28 activated browser scenarios passed on tablet/mobile.
- 28 desktop browser cases passed across split runs.
- 3 homepage model-story viewports passed.
- 13 API Markdown routes passed.
- Mobile AppHost API search passed.
- `git diff --check` passed.

No local production build was run.

### CODE REVIEW

File: src/frontend/tests/unit/custom-components.vitest.test.ts [hunk 1]
```diff
@@ -925,27 +925,36 @@ describe('custom Astro component render coverage', () => {
...
+        expect(tabsHtml).not.toContain(`data-supports-${language.id}`);
+      }
+    }
   });
 
-  it('removes disabled AppHost language pivots from rendered output', async () => {
+  it('renders AppHost language pivots according to the registry bit', async () => {
+    const pythonEnabled = appHostLanguageConfig.languages.find(
+      (language) => language.id === 'python'
+    )?.enabled;
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply src/frontend/tests/unit/custom-components.vitest.test.ts]: Replaced all optional Python lookups in this test file with the required getAppHostLanguage helper in 00f68740.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: src/frontend/tests/unit/apphost-languages.vitest.test.ts [hunk 2]
```diff
@@ -104,13 +105,30 @@ describe('AppHost language registry', () => {
     expect(normalizeAppHostLanguage('C#')).toBe('csharp');
     expect(normalizeAppHostLanguage('javascript')).toBeUndefined();
     expect(normalizeAppHostLanguage('mongo')).toBeUndefined();
-    expect(normalizeAppHostLanguage('python')).toBeUndefined();
+    expect(normalizeAppHostLanguage('python')).toBe(
+      appHostLanguageConfig.languages.find((language) => language.id === 'python')?.enabled
+        ? 'python'
+        : undefined
+    );
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply src/frontend/tests/unit/apphost-languages.vitest.test.ts:111]: Replaced the optional lookup with the existing throwing getAppHostLanguage helper in 00f68740, so a missing Python registry entry now fails the test explicitly.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: src/frontend/tests/unit/update-samples.vitest.test.ts [hunk 3]
```diff
@@ -142,14 +143,18 @@ describe('sample AppHost presentation metadata', () => {
...
+    'python',
+    'go',
+    'java',
+    'rust',
+  ] as const)('reflects registry activation for %s', (kind) => {
+    const languageId = kind === 'csproj' || kind === 'file-based' ? 'csharp' : kind;
+    const enabled = appHostLanguageConfig.languages.find(
+      (language) => language.id === languageId
+    )?.enabled;
     expect(isAppHostEnabled(kind as AppHostKind)).toBe(enabled);
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply src/frontend/tests/unit/update-samples.vitest.test.ts:156]: Replaced the generic optional lookup with getAppHostLanguage in 00f68740, so missing registry entries can no longer be treated as disabled.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

File: src/frontend/src/components/SimpleAppHostCode.astro [hunk 4]
```diff
@@ -21,11 +25,22 @@ const { lang = 'csharp', mark, collapse }: Props = Astro.props;
...
 ---
 
-{
-  typescriptCode ? (
-    <AppHostTabs>
+<AppHostTabs {limitations}>
+  {
+    typescriptCode && (
       <Fragment slot="typescript">
         <Code
```
Comment (mitchdenny) [inline src/frontend/src/components/SimpleAppHostCode.astro:45]: Please make the TypeScript slot genuinely absent when `typescriptCode` is undefined. Astro still registers this conditional named slot, so `AppHostTabs` sees `Astro.slots.has('typescript') === true`, emits `data-supports-typescript`, and ignores the supplied TypeScript limitation.

Reproduced at `/ja/get-started/app-host/?lang=go&aspire-lang=typescript` and `/ja/get-started/app-host/?lang=java&aspire-lang=typescript`: the selected TypeScript tabpanel is visible but empty, the C# example is hidden, and no TypeScript limitation is rendered.

Branch outside `AppHostTabs` so the C#-only branch does not declare a TypeScript slot, and add Go/Java regression tests asserting that the limitation appears rather than an empty tabpanel.

**Review by mitchdenny — CHANGES_REQUESTED** [review]
Found one rendering regression: C#-only SimpleAppHostCode examples now expose an empty TypeScript tab instead of the intended limitation. Reproduced for both Go and Java guest examples in the browser; details are inline.

Coverage: 182 targeted unit tests and 39 browser cases passed across desktop, tablet, and mobile, including homepage dark-theme axe checks. Those tests do not exercise the affected C#-only SimpleAppHostCode branch. No production build was run.

**Comment by mitchdenny** [comment]
The AppHost language selector looks uneven with all six languages enabled. At a 1440px desktop viewport, the buttons wrap into two rows, but `justify-content: space-between` spreads each row independently: Java sits at the far left of the second row and Rust at the far right, leaving a large empty gap between them.

Can we use a consistent grid or left-aligned wrapping with a fixed gap so the language options remain grouped and aligned across rows? This is in `src/frontend/src/components/PivotSelector.astro` (`.pivot-selector`).

Reproduced locally on the English `/get-started/app-host/?aspire-lang=python` page at commit `83dda4c4`, in dark mode. This layout feedback is separate from the earlier Japanese-page empty-tab finding.

AppHost language selector with Java and Rust spread across an otherwise empty second row

**Comment by alistairmatthews** [comment]
The "Select your AppHost language" component and the language selection tabs seem independent. E.g. this is from running this branch locally and browsing to the "What is the AppHost?" page. I've selected Python and Java at the same time. Shouldn't they be synced?
image
